packages/grid/src/vaadin-grid.js

```diff
@@ -546,6 +546,18 @@ class Grid extends ElementMixin(
       col.width = 'auto';
       col._origFlexGrow = col.flexGrow;
       col.flexGrow = 0;
+
+      if (col.parentElement != null && col.parentElement.localName === 'vaadin-grid-column-group') {
```

Comment:
This only takes the first parent group into account. With the following:

```html
<vaadin-grid>
  <vaadin-grid-column-group header="foo foo foo">
    <vaadin-grid-column-group header="foo foo">
      <vaadin-grid-column header="foo" auto-width flex-grow="0"></vaadin-grid-column>
    </vaadin-grid-column-group>
  </vaadin-grid-column-group>
</vaadin-grid>
```

...the outermost column group header gets clipped:

<img width="311" alt="Screenshot 2021-11-02 at 11 17 21" src="https://user-images.githubusercontent.com/1222264/139819241-807f548b-0aa2-400b-9b18-3b2ca76219c3.png">